### PR TITLE
JDK-8313670: Simplify shared lib name handling code in some tests

### PR DESCRIPTION
**Pull request by MBaesken**
There is coding e.g. in
https://github.com/openjdk/jdk/blob/master/test/jdk/jdk/jfr/event/runtime/TestNativeLibrariesEvent.java#L72
that deals with shared lib naming on different OS.
This code should be simplified.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313670](https://bugs.openjdk.org/browse/JDK-8313670): Simplify shared lib name handling code in some tests (**Bug** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [9f9c5b25](https://git.openjdk.org/jdk/pull/15151/files/9f9c5b259a2c789312abb94cc3ca385c501106e4)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [9f9c5b25](https://git.openjdk.org/jdk/pull/15151/files/9f9c5b259a2c789312abb94cc3ca385c501106e4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15151/head:pull/15151` \
`$ git checkout pull/15151`

Update a local copy of the PR: \
`$ git checkout pull/15151` \
`$ git pull https://git.openjdk.org/jdk.git pull/15151/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15151`

View PR using the GUI difftool: \
`$ git pr show -t 15151`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15151.diff">https://git.openjdk.org/jdk/pull/15151.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15151#issuecomment-1665361602)

### CODE REVIEW

File: test/lib/jdk/test/lib/Platform.java [hunk 1]
```diff
@@ -362,6 +362,18 @@ public static String sharedLibraryExt() {
...
+     * @return file name prefix
+     */
+    public static String sharedLibraryPrefix() {
+        if (isWindows()) {
+            return "";
+        } else {
+            return "lib";
+        }
+    }
+
```
Comment (plummercj) [inline test/lib/jdk/test/lib/Platform.java:376]: The following tests could leverage this new API. Just look for calls to `Platform.sharedLibraryExt()`:

test/hotspot/jtreg/runtime/signal/SigTestDriver.java
test/hotspot/jtreg/vmTestbase/nsk/jvmti/NativeLibraryCopier.java
test/jdk/com/sun/tools/attach/warnings/DynamicLoadWarningTest.java

Perhaps a `Platform.buildSharedLibraryName()` API is worth considering.

**Review by plummercj — CHANGES_REQUESTED** [review]

File: test/lib/jdk/test/lib/Platform.java (hunk 1 above)
Comment (sspitsyn) [reply test/lib/jdk/test/lib/Platform.java:376]: Agreed with suggestion from Chris.

Comment (dholmes-ora) [reply test/lib/jdk/test/lib/Platform.java:376]: Agreed, I'd like to see `Platform.buildSharedLibraryName()` in addition to the other methods (the later are needed if you want to decompose a filename to get the library name (though we could also provide a function just to do that).

**Comment by MBaesken** [comment]
Hello, as suggested I introduced a Platform.buildSharedLibraryName; this can be used to simplify a lot of code in the mentioned tests.

**Review by plummercj — APPROVED** [review]
Needs copyright updates, but otherwise looks good.

**Review by sspitsyn — APPROVED** [review]
Thank you for the update.
Looks good.
Thanks,
Serguei

**Comment by MBaesken** [comment]
Hi Chris and Serguei, thanks for the reviews !  I adjusted the COPYRIGHT years.

/integrate

**Review by dholmes-ora — COMMENTED** [review]
A belated thumbs up from me too. Thanks.

**Pull request closed without merging**